editor/plugins/version_control_editor_plugin.cpp

```diff
@@ -94,213 +125,774 @@ void VersionControlEditorPlugin::popup_vcs_set_up_dialog(const Control *p_gui_ba
+	diff->clear();
+	diff_content.clear();
+	diff_title->set_text("");
+}
+
+void VersionControlEditorPlugin::_item_activated(Object *p_tree) {
+	Tree *tree = Object::cast_to<Tree>(p_tree);
+
+	_move_item(tree, tree->get_selected());
+	_refresh_stage_area();
+}
 
-	_display_file_diff(file_path);
+void VersionControlEditorPlugin::_move_item(Tree *p_tree, TreeItem *p_item) {
+	CHECK_PLUGIN_INITIALIZED();
+
+	if (p_tree == staged_files) {
+		EditorVCSInterface::get_singleton()->unstage_file(p_item->get_meta("file_path"));
+	} else {
+		EditorVCSInterface::get_singleton()->stage_file(p_item->get_meta("file_path"));
+	}
 }
 
-void VersionControlEditorPlugin::_display_file_diff(String p_file_path) {
-	Array diff_content = EditorVCSInterface::get_singleton()->get_file_diff(p_file_path);
+void VersionControlEditorPlugin::_cell_button_pressed(Object *p_item, int p_column, int p_id) {
+	TreeItem *item = Object::cast_to<TreeItem>(p_item);
+	String file_path = item->get_meta("file_path");
+	EditorVCSInterface::ChangeType change = (EditorVCSInterface::ChangeType)(int)item->get_meta("change_type");
 
-	diff_file_name->set_text(p_file_path);
+	if (p_id == BUTTON_TYPE_OPEN && change != EditorVCSInterface::CHANGE_TYPE_DELETED) {
+		DirAccess *dir = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+		if (!dir->file_exists(file_path)) {
+			return;
+		}
+		memdelete(dir);
+
+		file_path = "res://" + file_path;
+		if (ResourceLoader::get_resource_type(file_path) == "PackedScene") {
+			EditorNode::get_singleton()->open_request(file_path);
+		} else if (file_path.ends_with(".gd")) {
+			EditorNode::get_singleton()->load_resource(file_path);
+			ScriptEditor::get_singleton()->reload_scripts();
+		} else {
+			EditorNode::get_singleton()->get_filesystem_dock()->navigate_to_path(file_path);
+		}
+
+	} else if (p_id == BUTTON_TYPE_DISCARD) {
+		_discard_file(file_path, change);
+		_refresh_stage_area();
+	}
+}
+
+void VersionControlEditorPlugin::_display_diff(int p_idx) {
+	DiffViewType diff_view = (DiffViewType)diff_view_type_select->get_selected();
 
 	diff->clear();
-	diff->push_font(EditorNode::get_singleton()->get_gui_base()->get_font("source", "EditorFonts"));
+
+	if (show_commit_diff_header) {
+		Dictionary meta_data = commit_list->get_selected()->get_metadata(0);
+		String commit_id = meta_data["commit_id"];
+		String commit_subtitle = meta_data["commit_subtitle"];
+		String commit_date = meta_data["commit_date"];
+		String commit_author = meta_data["commit_author"];
+
+		diff->push_font(EditorNode::get_singleton()->get_gui_base()->get_font("doc_bold", "EditorFonts"));
+		diff->push_color(EditorNode::get_singleton()->get_gui_base()->get_color("accent_color", "Editor"));
+		diff->add_text(TTR("Commit") + ": " + commit_id + "\n");
+		diff->add_text(TTR("Author") + ": " + commit_author + "\n");
+		diff->add_text(TTR("Date") + ": " + commit_date + "\n");
+		if (!commit_subtitle.empty()) {
+			diff->add_text(TTR("Subtitle") + ": " + commit_subtitle + "\n");
```

Comment:
```suggestion
		diff->add_text(TTR("Commit:") + " " + commit_id + "\n");
		diff->add_text(TTR("Author:") + " " + commit_author + "\n");
		diff->add_text(TTR("Date:") + " " + commit_date + "\n");
		if (!commit_subtitle.empty()) {
			diff->add_text(TTR("Subtitle:") + " " + commit_subtitle + "\n");
```

The colon should be part of the translated string so that it can be localized (for example French puts a space before colons, and CJK might have their own colon character).